### PR TITLE
Fix windows cuda build

### PR DESCRIPTION
**Pull request by jnnnnn**
* add ggml-cuda.cu
* sort out linker paths using CUDA_PATH

### CODE REVIEW

File: sys/build.rs [hunk 1]
```diff
@@ -34,11 +34,19 @@ fn main() {
     #[cfg(feature = "cuda")]
     {
         println!("cargo:rustc-link-lib=cublas");
-        println!("cargo:rustc-link-lib=culibos");
         println!("cargo:rustc-link-lib=cudart");
         println!("cargo:rustc-link-lib=cublasLt");
-        println!("cargo:rustc-link-search=/usr/local/cuda/lib64");
-        println!("cargo:rustc-link-search=/opt/cuda/lib64");
+        #[cfg(target_os = "windows")]
```
Comment (tazz4843) [inline sys/build.rs:39]: This should be changed to a cfg_if! block.

**Review by tazz4843 — CHANGES_REQUESTED** [review]
Minor nitpick, otherwise looks fine.

**Review by tazz4843 — APPROVED** [review]

**Comment by joseluis** [comment]
Thaks! this PR also fixes compilation in linux when enabling the `cuda` feature with whisper-rs as a dependency.

I was going to open a new issue, when I saw this already solved it. A new patch release of whisper-sys would be nice.

**Comment by tazz4843** [comment]
0.6.1 of whisper-rs-sys has been published, sorry that took me so long @joseluis

**Pull request merged**